quic/s2n-quic-platform/src/message/mmsg.rs

```diff
@@ -32,21 +27,18 @@ impl MessageTrait for mmsghdr {
 
     #[inline]
     fn payload_len(&self) -> usize {
-        self.msg_len as usize
+        let payload_len = self.msg_len as usize;
+        debug_assert!(payload_len <= core::u16::MAX as usize);
+        payload_len
     }
 
     #[inline]
     unsafe fn set_payload_len(&mut self, len: usize) {
-        debug_assert!(len <= core::u32::MAX as usize);
+        debug_assert!(len <= core::u16::MAX as usize);
```

Comment:
why are we going from u32 to u16. was the previous check not correct?

Reply:
the `msg_len` is a `u32`. but in general we don't allow payloads to be bigger that u16